### PR TITLE
Lodash: Refactor away from `_.omit()`

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR removes Lodash's `_.omit()` completely and deprecates the import of that function. There are a few remaining usages in that package and conversion is pretty straightforward. 

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

We're creating an inline helper function to reduce repetition. It will omit the unnecessary IDs that are passed as an array, converting the IDs specifically to numbers to ensure that they're compared properly (like Lodash did under the hood when calling `_.omit`).

I'm also updating docs that used `_.omit()` in a couple of examples and deprecating the import of `_.omit()` in our ESLint config.

## Testing Instructions

Verify all checks are still green. The changes are covered by unit tests:

```
npm run test:unit packages/core-data
```

### CODE REVIEW

File: packages/core-data/src/queried-data/reducer.js
Comment (Mamaduka) [inline packages/core-data/src/queried-data/reducer.js]: @tyxla, do you know if `omit` also casts values internally?

Not every entity ID will be an `int`. e.g., Templates and template parts use strings for IDs.

```
wp.data.select('core').getEntityRecords( 'postType', 'wp_template' );
```

Comment (tyxla) [reply packages/core-data/src/queried-data/reducer.js]: Yes, it does. It calls `toKey` on each value:

```
    function toKey(value) {
      if (typeof value == 'string' || isSymbol(value)) {
        return value;
      }
      var result = (value + '');
      return (result == '0' && (1 / value) == -INFINITY) ? '-0' : result;
    }
```

Thanks, @Mamaduka, I'll have to address your feedback and take strings into consideration there.

Comment (tyxla) [reply packages/core-data/src/queried-data/reducer.js]: Hey, @Mamaduka in https://github.com/WordPress/gutenberg/pull/46674/commits/26425eb433849692a37e956e1902147f9559f7c2 we're now supporting both ID and string IDs. I don't think we need to overcomplicate it to support other NaN values or symbols.

Please, take another look when you have a chance!

**Review by Mamaduka — APPROVED** [review]
This tests well for me. Manually tested on the screen where we can remove templates.

Considering the importance of the reducer, it would be nice to get one more review before merging the PR. Just in case we missed something :)

File: packages/core-data/src/queried-data/reducer.js
Comment (Mamaduka) [reply packages/core-data/src/queried-data/reducer.js]: Thanks for the quick follow-up. Would adding the test case for items with string IDs make sense here?

Comment (tyxla) [reply packages/core-data/src/queried-data/reducer.js]: I think that's a good idea @Mamaduka, thanks! Test that covers support of string IDs added in 8964b9c43e4c289af803ed137af456ba81e24d01

**Comment by tyxla** [comment]
Thanks, @Mamaduka!

I've also tested deleting a navigation menu entity from the Advanced settings of a Navigation block, so I'm pretty confident that this works well:

Screenshot 2022-12-21 at 10 52 39

**Pull request merged**
